lib/gc/ExecutionEngine/GPURuntime/ocl/GpuOclRuntime.cpp

```diff
@@ -993,6 +993,43 @@ OclModuleBuilder::build(const OclRuntime::Ext &ext) {
     mod.dump();
   }
 
+  if (spirvDump) {
+    mod->walk([&](LLVM::GlobalOp global) {
+      auto isaKernel = [&](LLVM::GlobalOp op) {
+        return op.getName().starts_with("gcGpuOclKernel_") &&
+               op.getName().ends_with("SPIRV");
+      };
+
+      if (!isaKernel(global))
+        return WalkResult::skip();
+
+      auto name = global.getName();
```

Comment:
```suggestion
      auto name = global.getName();
      
      if (!name.starts_with("gcGpuOclKernel_") && !name.ends_with("SPIRV") {
        return WalkResult::skip();
      }
```

I think, lambda is redundant here, it's called only once.